### ISSUE TITLE
My Home: remove the "Site created" checklist item

### ISSUE THREAD

**Issue by autumnfjeld** [Automattic/wp-calypso#53138]
### Overview
This is part of [My Home 🏠 Pod short-term design improvements](pd2qGl-2P-p2#short-term-goals)

My Home: Remove the "Site created" checklist item
![image](https://user-images.githubusercontent.com/5665959/119479368-e8e38880-bd93-11eb-9c37-e4ed823369b7.png)

### User facing acceptance criteria
User has gone through site onboarding
When they land in My Home 
Then they see the Site Checklist they DO NOT see a "Site created" checklist item

### Figma Designs
N/A for this ticket

### Related links
[My Home Project thread]( pd2qGl-17-p2)
[My Home: Short-term design improvements](pd2qGl-4i-p2)
[Figma for related work](https://href.li/?https://www.figma.com/file/fLQcNzdadWzo87n89Imecj/?node-id=1763%3A0)

### PR that fixes issue

**Comment by p-jackson** [comment]
I've implemented this change in D62160-code

The checklist now looks like this when creating a new site.

https://user-images.githubusercontent.com/1500769/120427656-64fa5380-c3c6-11eb-9416-25791913296c.mov



If we decided (for whatever reason) to revert this change then we might want to be careful to only put it back in place for new sites. The existing code had logic to only show the task for sites created after the date it was introduced, so that introducing it wouldn't require users to go back and check off a task that wasn't really a task. It'd be easier to revert the above change and just adjust the date to when we reverted it.
